File: LeetCode/Longest Subarray with Maximum Bitwise AND.ts
function longestSubarray(nums: number[]): number {
  const max_element = Math.max(...nums);

  let count = 0,
    max = 0;

  for (const num of nums) {
    if (num == max_element) count++;
    else count = 0;

    max = Math.max(max, count);
  }

  return max;
}

File: LeetCode/Longest Subarray with Maximum Bitwise AND.go
func longestSubarray(nums []int) int {
    max_element := math.MinInt
    current, ans := 0, 0
    
    for i:=0; i < len(nums); i++ {
        if nums[i] > max_element {
            max_element = nums[i]
        }
    }
    
    for i:=0; i < len(nums); i++ {
        if (nums[i] == max_element) {
            current++
        } else {
            current = 0
        }
        
        if (current > ans) {
            ans = current
        }
    }
    
    return ans
}